src/Scrollyteller/index.tsx

```diff
@@ -97,6 +67,46 @@ const Scrollyteller = (props: Props) => {
     return () => {
       window.removeEventListener('scroll', onScroll);
     };
+
+    function onScroll(_event: any, dontFireInitialMarker?: boolean) {
+      const { config } = props;
+
+      if (references.length === 0) return;
+
+      // Work out which panel is the current one
+      const fold = window.innerHeight * (config.waypoint ? config.waypoint / 100 : 0.8);
+      const referencesAboveTheFold = references.filter((r: any) => {
+        if (!r.element) return false;
+        const box = r.element.getBoundingClientRect();
+        return box.height !== 0 && box.top < fold;
+      });
+
+      let closestReference = referencesAboveTheFold[referencesAboveTheFold.length - 1];
+      if (!closestReference) closestReference = references[0];
+
+      if (currentPanel !== closestReference.panel) {
+        currentPanel = closestReference.panel;
+        if (!dontFireInitialMarker) {
+          onMarkerRef.current(closestReference.panel.config, closestReference.panel.id);
```

Comment:
Moving onScroll inside useEffects changes. The only real change is this line, changed from `onMarker(closestReference.panel.config, closestReference.panel.id);`